src/Microsoft.Graph.Core/Exceptions/Error.cs

```diff
@@ -12,15 +12,35 @@ namespace Microsoft.Graph
 
     /// <summary>
     /// The error object that handles unsuccessful responses returned from the service.
+    /// Models OData protocol, 9.4 Error Response Body
+    /// http://docs.oasis-open.org/odata/odata/v4.01/csprd05/part1-protocol/odata-v4.01-csprd05-part1-protocol.html#_Toc14172757
     /// </summary>
     [JsonObject(MemberSerialization = MemberSerialization.OptIn)]
     public class Error
     {
         /// <summary>
-        /// The HTTP status code.
+        /// This code serves as a sub-status for the HTTP error code specified in the response.
```

Comment:
The code property for the Error object at the root of the document is a 1:1 mapping to the HTTP status code.  When the Error object is in InnerError or ErrorDetails then the code becomes a sub-code.

Reply:
We should sync with Saurabh as they are currently making changes to ODataLib in this area and we should try and align.  /cc @madansr7

Reply:
According to spec, it is supposed to a sub status code:

*code*: required non-null, non-empty, language-independent string. Its value is a service-defined error code. This code serves as a sub-status for the HTTP error code specified in the response.

In actuality, I recall seeing it either represent the actual HTTP error code in some cases, in other cases it is a workload provided code, and in other cases, our client sets it as Unknown. I should change the comment to be what it actual represents, not what it is supposed to. :(

Reply:
@MIchaelMainer LGTM! What was the conclusion on code property for Error object? /cc @darrelmiller

Reply:
@MIchaelMainer   Hmm, weird, I thought I replied to this.  The top level code should never be a workload specific code moving forward.

Reply:
We need to change this comment.  We need to clarify that the value of the code depends on whether it is a top level Error or an innerError.

Reply:
Updated.